yt_dlp/extractor/zingmp3.py

```diff
@@ -1,22 +1,51 @@
 # coding: utf-8
 from __future__ import unicode_literals
 
+import hashlib
+import hmac
+from urllib.parse import urlencode
+
 from .common import InfoExtractor
 from ..utils import (
     int_or_none,
+    try_get,
+    traverse_obj,
 )
 
 
 class ZingMp3BaseIE(InfoExtractor):
-    _VALID_URL_TMPL = r'https?://(?:mp3\.zing|zingmp3)\.vn/(?:%s)/[^/]+/(?P<id>\w+)\.html'
+    _VALID_URL_TMPL = r'https?://(?:mp3\.zing|zingmp3)\.vn/(?P<type>(?:%s))/[^/]+/(?P<id>\w+)(?:\.html|\?)'
     _GEO_COUNTRIES = ['VN']
+    _IS_UPDATE_COOKIES = False
+    _DOMAIN = 'https://zingmp3.vn'
+    _SLUG_API = {
+        'bai-hat': '/api/v2/page/get/song',
+        'embed': '/api/v2/page/get/song',
+        'video-clip': '/api/v2/page/get/video',
+        'playlist': '/api/v2/page/get/playlist',
+        'album': '/api/v2/page/get/playlist',
+        'lyric': '/api/v2/lyric/get/lyric',
+        'song_streaming': '/api/v2/song/get/streaming',
+    }
+
+    _TEMP_TIMESTAMP = '123456'
```

Comment:
what is this?

Reply:
@Ashish0804 As i mentioned in the comment above, this timestamp is just a string to put in the hash, so maybe I'll hardcode it a sample text.

Reply:
Don't make it as a variable then imo.
I mean assuming it's working with not even an valid timestamp, just hardcode the text directly.

Reply:
Yes, let me change.